app/models/user.rb

```diff
@@ -118,4 +118,8 @@ def ssn_last_four
   def opted_into_notifications?
     sms_notification_opt_in_yes? || email_notification_opt_in_yes?
   end
+
+  def age_end_of_tax_year
+    intake.tax_year - Date.parse(birth_date).year if birth_date
```

Comment:
a style nitpick: I find the `if birth_date` guard clause combined with the implicit return to be a bit nonintuitive due to the low associativity of `if` by the ruby parser. Mentally, I parse the statement as

```ruby
intake.tax_year - (Date.parse(birth_date).year if birth_date)
```
which is not how ruby does it (and doesn't work). I think it's more clear to explicitly return for cases we aren't expecting the math to work:

```ruby
def age_end_of_tax_year
  return unless birth_date.present?

  intake.tax_year - Date.parse(birth_date).year
end
```

Do you agree?

Reply:
i think that looks better! the `if birth_date` was a quick fix for a broken test haha

what is "low associativity"?

Reply:
Ah, I might have used the wrong word, "precedence" might be the right one. I was going for the concept explained here: https://www.tinfoilsecurity.com/blog/ruby-demystified-and-vs except for "if" rather than "and"/"or".

It's a fancy way to describe the concept of where parentheses are automatically added to an expression. For example

```ruby
> a = true && false      # parsed as: a = (true && false)
> a
true

> b = true and false      # parsed as: (b = true) && false
> b
false
```

Reply:
ohhh got it - thanks!